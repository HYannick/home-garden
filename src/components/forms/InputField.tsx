/** @jsx jsx */
import React, {Fragment} from 'react';
import styled from "@emotion/styled";
import {jsx} from '@emotion/core';
import {Label} from './Label';

interface FieldProps {
  field?: any,
  form?: any,
  placeholder?: string
  label?: string,
  alignment?: string,
  type: string
}

const Input = styled('input')<{ alignment: string }>`
  border: 0.3rem solid ${({theme}) => theme.palette.grey.dark};
  outline: ${({theme}) => theme.palette.grey.light};
  padding: 1rem 2rem;
  text-align: ${({alignment}) => alignment || 'left'};
  border-radius: 1.5rem;
  width: 100%;
  font-size: 1.6rem;
`;


const InputField: React.FC<FieldProps> = ({field, label, alignment, type}) => (
  <Fragment>
    <Label htmlFor={field.name}>{label}</Label>
    <Input id={field.name} name={field.name} type={type} onChange={field.onChange} alignment={alignment || ''}/>
  </Fragment>
);

export default InputField;
