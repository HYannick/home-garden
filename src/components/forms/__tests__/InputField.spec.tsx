import React from 'react';
import 'jest-dom/extend-expect';
import { cleanup, fireEvent } from '@testing-library/react';
import { renderWithTheme } from '../../../theme-wrapper';
import InputField from '../InputField';

afterEach(cleanup);
describe('InputField', () => {
  const props = {
    field: {
      name: 'puke',
      onChange: jest.fn(),
    },
    type: 'text',
    label: 'Should I puke?',
  };
  it('should render properly', () => {
    const { container } = renderWithTheme(
      <InputField {...props}/>,
    );
    expect(container).toMatchSnapshot();
  });
  it('should render the right label', () => {
    const { getByText } = renderWithTheme(
      <InputField {...props}/>,
    );
    expect(getByText(/^Should I puke/)).toHaveTextContent('Should I puke');
  });
  it('should trigger onChange', () => {
    const { getByLabelText } = renderWithTheme(
      <InputField {...props}/>,
    );
    const input = getByLabelText('Should I puke?');
    fireEvent.change(input, { target: { value: 'm' } });
    expect(props.field.onChange).toHaveBeenCalledTimes(1);
  });
});
