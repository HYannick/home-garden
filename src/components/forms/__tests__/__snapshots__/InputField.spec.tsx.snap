// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField should render properly 1`] = `
<div>
  <label
    class="css-npa6bc"
    for="puke"
  >
    Should I puke?
  </label>
  <input
    class="css-1qxlqm1"
    id="puke"
    name="puke"
    type="text"
    value=""
  />
</div>
`;
